src/server/common/controller/page-controller/page-controller.js

```diff
@@ -0,0 +1,61 @@
+import { calculateNextPage } from '../../helpers/next-page.js'
+import { ExitPage } from '../../model/page/exit-page-model.js'
+/** @import { Server, ServerRegisterPluginObject } from '@hapi/hapi' */
+/** @import { NextPage } from '../../helpers/next-page.js' */
+/** @import { RawPayload } from '../../model/answer/answer-model.js' */
+/** @import { Page } from '../../model/page/page-model.js' */
+
+export class PageController {
+  /**
+   * @param {Page} page
+   */
+  constructor(page) {
+    this.page = page
+  }
+
+  /** @returns {ServerRegisterPluginObject<void>} */
+  plugin() {
+    return {
+      plugin: {
+        name: `${this.page.sectionKey}-${this.page.key}`,
+
+        /** @param {Server} server */
+        register: (server) => {
+          server.route([
+            {
+              method: 'GET',
+              path: this.page.urlPath,
+              handler: this.getHandler.bind(this)
+            },
+            {
+              method: 'POST',
+              path: this.page.urlPath,
+              handler: this.postHandler.bind(this)
+            }
+          ])
+        }
+      }
+    }
+  }
+
+  getHandler(req, h) {
+    return h.view(this.page.view, {
+      nextPage: req.query.redirect_uri,
+      pageTitle: this.page.title,
+      heading: this.page.heading,
+      hideQuestion: true,
+      ...this.page.viewProps
+    })
+  }
+
+  postHandler(req, h) {
+    const payload = /** @type {NextPage} */ (req.payload)
+    const nextPage = this.page.nextPage()
+
+    if (nextPage instanceof ExitPage) {
+      return h.redirect(nextPage.urlPath)
+    } else {
+      return h.redirect(calculateNextPage(payload.nextPage, nextPage.urlPath))
+    }
```

Comment:
Hrm - I'm guessing for this ticket, it's not going to be a POST - but a straightforward GET?

I'm not 100% sure which page type we'd use this in.

Reply:
As discussed on slack, we can add a hook in here for an optional POST controller.  I don't think that we should add in a POST controller if it's not needed - since:

- it's a bit misleading to the viewer of the code - e.g. they'll think it's being used
- it's adding something to the customer-facing journey (e.g. a member of the public can post against the server), even though it doesn't make much sense for this use case.

Reply:
as mentioned - this is for the interstitial page found [here](https://applicationsandpermissions-dev-ffeda85e15b1.herokuapp.com/v1_1_0/create-application/contact-and-updates/licence-select-post-can-not-use-this-service) which will need a post handler - conditional could work though.....